src/System.Web.OData/OData/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -82,6 +95,32 @@ private IQueryable Bind(IQueryable queryable)
             return selectMethod.Invoke(null, new object[] { queryable, projectionLambda }) as IQueryable;
         }
 
+        private Delegate GetCompiledProjectionLambda()
+        {
+            string cacheKey = String.Format("{0}?select={1}&$expand={2}", _selectExpandQuery.Context.ElementClrType.FullName, _selectExpandQuery.RawSelect, _selectExpandQuery.RawExpand);
+            object cachedValue = DelegateCache.Get(cacheKey);
+            if (cachedValue != null)
+            {
+                return (Delegate)cachedValue;
+            }
+
+            LambdaExpression expression = GetProjectionLambda();
+            Delegate compiledExpression = expression.Compile();
+
+            if (_settings.SelectExpandCacheExpirationTimeSeconds > 0)
+            {
+                // Use sliding window expiration to prevent unused things from hanging around indefinitely.
+                CacheItemPolicy evictionPolicy = new CacheItemPolicy
+                {
+                    SlidingExpiration = TimeSpan.FromSeconds(_settings.SelectExpandCacheExpirationTimeSeconds)
+                };
+
+                DelegateCache.Add(cacheKey, compiledExpression, evictionPolicy);
+            }
+
+            return compiledExpression;
```

Comment:
why is this break out of get/add more efficient than the MemoryCache's own AddOrGet method?